src/preprocessing/passes/non_clausal_simp.cpp

```diff
@@ -280,34 +282,19 @@ PreprocessingPassResult NonClausalSimp::applyInternal(
   // r' another constant propagation, then l'[l/r] -> r' should be a
   //    constant propagation too
   // 4. each lhs of constantPropagations is different from each rhs
-  for (pos = newSubstitutions.begin(); pos != newSubstitutions.end(); ++pos)
+  for (pos = nss.begin(); pos != nss.end(); ++pos)
```

Comment:
```suggestion
  for (const auto& pos: nss)
```
And replace `(*pos)` by `pos`.